### PR TITLE
Alyapunov/gh 5515 some mvcc crash

### PR DESCRIPTION
**Pull request by alyapunov**
Subj

This patch only fixes crashed in #5515. We must not close the issue after this merge.

### CODE REVIEW

File: test/box/tx_man.test.lua [hunk 1]
```diff
@@ -615,6 +615,16 @@ tx1:commit()
...
 
+-- https://github.com/tarantool/tarantool/issues/5515
+s = box.schema.space.create('test')
+i0 = s:create_index('pk', {parts={{1, 'uint'}}})
+i1 = s:create_index('i1', {id = 10, type = 'tree', parts={{2, 'uint'}}})
+i2 = s:create_index('i2', {id = 20, type = 'hash', parts={{2, 'uint'}}})
+i3 = s:create_index('i3', {id = 30, type = 'bitset', parts={{3, 'uint'}}})
+i4 = s:create_index('i4', {id = 40, type = 'rtree', parts={{4, 'array'}}})
+s:replace{1, 1, 15, {0, 0}}
+s:drop()
```
Comment (Korablev77) [inline test/box/tx_man.test.lua:638]: Test executes fine on master branch

**Review by Korablev77 — CHANGES_REQUESTED** [review]
CI is red; there are also conflicts with master branch...

**Comment by Korablev77** [comment]
These patches also fix https://github.com/tarantool/tarantool/issues/6137

File: test/box/tx_man.test.lua (hunk 1 above)
Comment (alyapunov) [reply test/box/tx_man.test.lua:685]: True. added 100% test

**Comment by alyapunov** [comment]
> CI is red; there are also conflicts with master branch...

fixed

**Review by Korablev77 — APPROVED** [review]
LGTM

**Comment by Korablev77** [comment]
Pushed also to 2.8 and 2.7

**Pull request merged**